src/main/java/org/broadinstitute/hellbender/engine/GATKTool.java

```diff
@@ -574,6 +588,34 @@ private void validateSequenceDictionaries() {
             }
         }
 
+        // Check the master dictionary against the reference / reads / features
+        if (masterSequenceDictionary != null) {
+
+            // Check against the reads
+            if (hasReads()) {
+                SequenceDictionaryUtils.validateDictionaries("sequence", masterSequenceDict, "reads", readDict);
+            }
+
+            // Check against the reference
+            if (hasReference()) {
+                SequenceDictionaryUtils.validateDictionaries("sequence", masterSequenceDict, "reads", refDict);
+            }
+
+            // Check against the features
+            for (final SAMSequenceDictionary featureDict : featureDicts) {
+                SequenceDictionaryUtils.validateDictionaries("sequence", masterSequenceDict, "features", featureDict);
+            }
+
+            // When overriding the master sequence dictionary and working with CRAM files, we need to make sure that the
+            // reference we're using is completely contained by the sequence dictionary (i.e. the sequence dictionary is
+            // equal to the reference OR the sequence dictionary is a superset of the reference).
+            // This is accomplished by the call to validateCRAMDictionaryAgainstReference.
+            if ( hasCramInput() ) {
+                SequenceDictionaryUtils.validateCRAMDictionaryAgainstReference(masterSequenceDict, refDict);
+            }
```

Comment:
The cram check needs to be moved up inside the `if (hasReads()) {` block. 

Ie., if you have reads, then if you have a cram input do the strict check, otherwise do the standard check).